### PR TITLE
ResStock-HPXML: Unit-level arguments for SFA/MF

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Addresses #728.
Companion PR: https://github.com/NREL/resstock-estimation/pull/176

Related to: https://github.com/NREL/OpenStudio-HPXML/pull/863

In `ResStockArguments`, exclude arguments from `BuildResidentialHPXML`:
* `geometry_unit_left_wall_is_adiabatic`
* `geometry_unit_right_wall_is_adiabatic`
* `geometry_unit_back_wall_is_adiabatic`
* `geometry_unit_front_wall_is_adiabatic`
* `geometry_unit_num_floors_above_grade`

Adds arguments to `ResStockArguments`:
* `geometry_unit_level`
* `geometry_unit_horizontal_location`
* `geometry_num_floors_above_grade`
* `geometry_corridor_position`
* `geometry_corridor_width`

Then in `ResStockArguments` a mapping is made from new `ResStockArguments` to excluded `BuildResidentialHPXML`.

## Checklist

Not all may apply:

- [ ] Unit tests have been added or updated
- [ ] All rake tasks have been run, and pass
- [ ] Documentation has been modified appropriately
- [ ] Any new options are added to `project_testing`
- [ ] `project_testing` runs without any failures
- [ ] No unexpected regression test changes
- [ ] All tests are passing
- [ ] The [changelog](https://github.com/NREL/resstock/blob/develop/CHANGELOG.md) has been updated appropriately
- [ ] This branch is up-to-date with develop

For more information on how to perform these checklist items, see the documentation's [Advanced Tutorial](https://resstock.readthedocs.io/en/latest/advanced_tutorial/index.html).

### CODE REVIEW

File: resources/hpxml-measures/BuildResidentialHPXML/measure.rb [hunk 1]
```diff
@@ -3493,7 +3493,7 @@ def self.set_air_infiltration_measurements(hpxml, runner, args)
 
   def self.set_roofs(hpxml, runner, model, args, sorted_surfaces)
     args[:geometry_roof_pitch] *= 12.0
-    if args[:geometry_attic_type] == HPXML::AtticTypeFlatRoof
+    if args[:geometry_attic_type] == HPXML::AtticTypeFlatRoof or args[:geometry_attic_type] == HPXML::AtticTypeBelowApartment
```
Comment (shorowit) [inline resources/hpxml-measures/BuildResidentialHPXML/measure.rb:3496]: Does this actually do anything? A MF unit that is below another unit shouldn't end up with a HPXML `Roof` (which is always adjacent to outside) but rather a HPXML `FrameFloor` with `ExteriorAdjacentTo="other housing unit"`.

Comment (aspeake) [reply resources/hpxml-measures/BuildResidentialHPXML/measure.rb]: @shorowit `geometry_roof_pitch` can still affect the pitch of a PV system, which is why this fix was introduced.

**Comment by shorowit** [comment]
@aspeake Any idea why there are pretty big changes to upgrade savings on the CI for project_national (but not project_testing)? For example, here's natural gas heating:

image

**Comment by aspeake** [comment]
> 
> 
> @aspeake Any idea why there are pretty big changes to upgrade savings on the CI for project_national (but not project_testing)? For example, here's natural gas heating:
> 
> image

@shorowit this occurs because the branches have different walls adjacent to the corridor in the baseline building, essentially rotating the orientation 180 degrees. The building sampled is a MF building with a double-loaded corridor, which now assigns the front wall to wherever the door is (ie, the wall adjacent to the corridor). Previously, the double loaded corridor was the back wall.

**Review by shorowit — APPROVED** [review]
Looks reasonable.

**Pull request merged**